src/State/SerializerAwareProviderInterface.php

```diff
@@ -19,6 +19,8 @@
  * Injects serializer in providers.
  *
  * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ *
+ * @deprecated in 4.2, to be removed in 5.0 because it violates the dependency injection principle.
  */
 interface SerializerAwareProviderInterface
```

Comment:
add a trigger_deprecation on top of this interface?

Reply:
Done [here](https://github.com/api-platform/core/pull/7348/files#diff-819587065b2fe4a4ac1258f043969f82a1d3510302e3bd89e1ffb04ef4e404dbR34), it's not enough?